dist/utils/time_unit_utils.ts

```diff
@@ -23,11 +27,81 @@ export class TimeUnitUtils {
         return TimeUnit[unit];
     }
 
+    public static ceilingToAvailableUnit(interval: string, availableUnits: UnitValue[]): [string, string] {
+      const intervalMillis = this.intervalToMillis(interval);
+
+      // match the first available unit, available Units must be ordered
```

Comment:
Is something validating or sorting these when the configuration is specified?

Reply:
The code sorts the list after it converts it.  There is handling for bad data where it eliminates values that don't map to a unit/value.  I added tests to validate that behavior.